src/alert/alert.ts

```diff
@@ -38,12 +43,22 @@ export class NgbAlert {
   /**
    * An event emitted when the close button is clicked. This event has no payload. Only relevant for dismissible alerts.
    */
-  @Output() close = new EventEmitter();
+  @Output() close = new EventEmitter<void>();
 
-  constructor(config: NgbAlertConfig) {
+  constructor(config: NgbAlertConfig, private _renderer: Renderer2, private _element: ElementRef) {
     this.dismissible = config.dismissible;
     this.type = config.type;
   }
 
   closeHandler() { this.close.emit(null); }
+
+  ngOnChanges(changes: SimpleChanges) {
+    const {type} = changes;
+    if (type && !type.firstChange) {
+      this._renderer.removeClass(this._element.nativeElement, `alert-${type.previousValue}`);
```

Comment:
Why do we do this through low-level DOM manipulation? Couldn't we move the existing bindings to the host?
Or is binding overriding classes set by a user on a host?

Reply:
Double-checked, and it overrides user-set classes. So I see three options:

1. Leave as it is done currently with `Renderer2`
2. hardcode all supported classes and remove the demo with `alert-customclass`:

```
  host: {
    'role': 'alert',
    'class': 'alert d-block',
    '[class.alert-dismissible]': 'dismissible',
    '[class.alert-success]': `type === 'success'`,
    '[class.alert-warning]': `type === 'warning'`,
    '[class.alert-info]': `type === 'info'`,
    '[class.alert-danger]': `type === 'danger'`,
    '[class.alert-primary]': `type === 'primary'`,
    '[class.alert-secondary]': `type === 'secondary'`,
    '[class.alert-light]': `type === 'light'`,
    '[class.alert-dark]': `type === 'dark'`
  },
```
3. do nothing and close this :)